spec/index.bs

```diff
@@ -1514,169 +1518,160 @@ interface GPUMapMode {
-
-        Otherwise, let |rangeSize| be |size|.
-
-    1. If this call doesn't follow [$mapAsync Valid Usage$]:
-
-      1. Record a validation error on the current scope.
-      1. Return [=a promise rejected with=] an {{AbortError}} on the [=Device timeline=].
-
-    1. Let |p| be a new {{Promise}}.
-    1. Set |this|.{{GPUBuffer/[[mapping]]}} to |p|.
-    1. Set |this|.{{GPUBuffer/[[state]]}} to [=buffer state/mapping pending=].
-    1. Set |this|.{{GPUBuffer/[[map_mode]]}} to |mode|.
-    1. Enqueue an operation on the default queue's [=Queue timeline=] that will execute the following:
-
-        1. If |this|.{{GPUBuffer/[[state]]}} is [=buffer state/mapping pending=]:
-
-            1. Let |m| be a new {{ArrayBuffer}} of size |rangeSize|.
-            1. Set the content of |m| to the content of |this|'s allocation starting at offset |offset| and for |rangeSize| bytes.
-            1. Set |this|.{{GPUBuffer/[[mapping]]}} to |m|.
-            1. Set |this|.{{GPUBuffer/[[state]]}} to [=buffer state/mapped=].
-            1. Set |this|.{{GPUBuffer/[[mapping_range]]}} to <code>[|offset|, |offset| + |rangeSize|]</code>.
-            1. Set |this|.{{GPUBuffer/[[mapped_ranges]]}} to `[]`.
-            1. Resolve |p|.
-
-    1. Return |p|.
-
-    <div algorithm class=validusage>
-        <dfn abstract-op>mapAsync Valid Usage</dfn>
-
-        Given a {{GPUBuffer}} |this|, a {{GPUMapModeFlags}} |mode|, a {{GPUSize64}} |offset|
-        and a {{GPUSize64}} |rangeSize|, the following validation rules apply:
-
-        1. |this| must be a [=valid=] {{GPUBuffer}}.
-        1. |offset| must be a multiple of 4.
-        1. |rangeSize| must be a multiple of 4.
-        1. |offset| + |rangeSize| must be less or equal to |this|.{{[[size]]}}
-        1. |this|.{{GPUBuffer/[[state]]}} must be [=buffer state/unmapped=]
-        1. |mode| must contain exactly one of {{GPUMapMode/READ}} or {{GPUMapMode/WRITE}}.
-        1. If |mode| contains {{GPUMapMode/READ}} then |this|.{{[[usage]]}} must contain {{GPUBufferUsage/MAP_READ}}.
-        1. If |mode| contains {{GPUMapMode/WRITE}} then |this|.{{[[usage]]}} must contain {{GPUBufferUsage/MAP_WRITE}}.
-
-        Issue: Do we validate that |mode| contains only valid flags?
-    </div>
-</div>
-
-### <dfn method for=GPUBuffer>getMappedRange(offset, size)</dfn> ### {#GPUBuffer-getMappedRange}
+<dl dfn-type=method dfn-for=GPUBuffer>
+    : <dfn>mapAsync(mode, offset, size)</dfn>
+    ::
 
-<div algorithm="GPUBuffer.getMappedRange">
-    <strong>|this|:</strong> of type {{GPUBuffer}}.
+        <div algorithm=GPUBuffer.mapAsync>
+            **Called on:** {{GPUBuffer}} |this|.
 
-    **Arguments:**
+            **Arguments:**
+            <pre class=argumentdef for="GPUBuffer/mapAsync(mode, offset, size)">
+                |mode|:
+                |offset|:
+                |size|:
+            </pre>
 
-    - {{GPUSize64}} |offset|
-    - {{GPUSize64}} |size|
+            **Returns:** Promise<void>
```

Comment:
Seems slightly hazardous to have bare angle brackets in a place that looks like a tag. Maybe this should be
```suggestion
            **Returns:** `Promise<void>`
```